modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/FileParsers.js

```diff
@@ -69,48 +70,67 @@ export function extractFieldsFromCSV(
 ) {
 	const splitLines = content.split('\n');
 	const newLineFound = content.indexOf('\n') > -1;
+	let contentItemDetails;
 
 	if (csvContainsHeaders && splitLines.length > 2) {
 		const [schema, firstItemData] = parseCSV(content, csvSeparator);
+		contentItemDetails = parseCSV(content, csvSeparator).slice(
+			1,
+			content.length
+		);
+
+		Liferay.fire(FILE_FORMATTED_CONTENT, {
+			fileContent: contentItemDetails,
+		});
 
 		return {
+			contentItemDetails,
 			firstItemDetails: getItemDetails(firstItemData, schema),
 			schema,
 		};
 	}
 
 	if (!csvContainsHeaders && newLineFound) {
 		const [firstItemData] = parseCSV(splitLines[0], csvSeparator);
+		contentItemDetails = parseCSV(content, csvSeparator);
 
 		const schema = new Array(firstItemData.length)
 			.fill()
 			.map((_, index) => index);
 
+		Liferay.fire(FILE_FORMATTED_CONTENT, {
+			fileContent: contentItemDetails,
+		});
+
 		return {
+			contentItemDetails,
 			firstItemDetails: getItemDetails(firstItemData, schema),
 			schema,
 		};
 	}
 }
 
 export function extractFieldsFromJSONL(content) {
-	let contentToParse;
-
-	if (content.indexOf('\n') > -1) {
-		const splitLines = content.split('\n');
+	const contentLines = content.replace(/\r?\n/g, ',');
```

Comment:
Here I spotted a regression, sorry if I missed it before.

This function can receive a partial file content, which means that content can contain:
`'{"id": 123, "name": "Giulio"}\n{"id": 234. "nam'`

In your code this it what will happen:
1. the `\n` will be replaced by a `,`
2. the string will be wrapped within brackets
3. JSON.parse will receive `'[{"id": 123, "name": "Giulio"},{"id": 234. "nam']` as a parameter 
4. the application breaks

Please ensure to pass correct data to the function so it won't fail